modules/light-clients/08-wasm/types/wasm.go

```diff
@@ -1,61 +1,38 @@
 package types
 
 import (
-	"bytes"
-	"slices"
-
-	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/internal/ibcwasm"
 )
 
-// GetCodeHashes returns all the code hashes stored.
-func GetCodeHashes(ctx sdk.Context, cdc codec.BinaryCodec) (CodeHashes, error) {
-	wasmStoreKey := ibcwasm.GetWasmStoreKey()
-	store := ctx.KVStore(wasmStoreKey)
-	bz := store.Get([]byte(KeyCodeHashes))
-	if len(bz) == 0 {
-		return CodeHashes{}, nil
-	}
-
-	var hashes CodeHashes
-	err := cdc.Unmarshal(bz, &hashes)
+// GetAllCodeHashes is a helper to get all code hashes from the store.
+// It returns an empty slice if no code hashes are found
+func GetAllCodeHashes(ctx sdk.Context) ([][]byte, error) {
+	iterator, err := ibcwasm.CodeHashes.Iterate(ctx, nil)
```

Comment:
should we close the iterator? 

`defer iterator.Close()` or something?

Reply:
`iter.Keys` should close

Reply:
nice!